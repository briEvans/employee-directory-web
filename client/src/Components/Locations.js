import React, { Component } from 'react';

// Internal Routes
import '../App.scss';

class Locations extends Component {

  render() {
    return (
      <div className="app-body">
        <div className="container">
          <div className="header"><h2>Listing {this.props.category}</h2></div>
          <div className="employees">
            {
              this.props.locations.length > 0 ? (
                this.props.locations.map(item => {
                  return (
                    <div className="location">
                      <img className="location-image" src={item.image} alt={item.name} />
                      <h3>{item.name}</h3>
                      <span className="location-detail">Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                           Lorem Ipsum has been the industry's standard dummy text ever since the</span>
                    </div>
                  );
                })
              ) :(<div className="employees"><h4> There are no Locations</h4></div>)
            }
          </div>
        </div>
      </div>
    );
  }
};

export default Locations;
