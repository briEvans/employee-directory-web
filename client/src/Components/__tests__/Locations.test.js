import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route} from 'react-router-dom';
import { mount, shallow, render } from 'enzyme';

import Locations from '../Locations';

const baseProps = {
  'locations': [
    {
      "id": 1,
      "name": "Austin, Texas",
      "image": "https://www.mymilestone.com/blog/wp-content/uploads/2017/12/forecast-1.jpg"
    },
    {
      "id": 2,
      "name": "San Jose, California",
      "image": "https://www.valleyinnsanjose.com/gallery-images/properties/5/0/1/1540545718_attractions_1.jpg"

    },
    {
      "id": 3,
      "name": "Tokyo, Japan",
      "image": "https://www.telegraph.co.uk/content/dam/Travel/Destinations/Asia/Japan/Tokyo/tokyo-city-scape.jpg?imwidth=450"
    }
  ]
};

describe("Locations component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    <Router>
      ReactDOM.render(<Locations
        {...baseProps}/>, div);
    </Router>
    ReactDOM.unmountComponentAtNode(div);
  });

  test("it matches the snapshot", () => {
    const component = shallow(<Locations
      {...baseProps}/>);
    expect(component).toMatchSnapshot();
  });
});
