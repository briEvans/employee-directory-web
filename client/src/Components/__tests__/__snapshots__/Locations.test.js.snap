// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations component it matches the snapshot 1`] = `
<div
  className="app-body"
>
  <div
    className="container"
  >
    <div
      className="header"
    >
      <h2>
        Listing 
      </h2>
    </div>
    <div
      className="employees"
    >
      <div
        className="location"
        key="Austin, Texas"
      >
        <img
          alt="Austin, Texas"
          className="location-image"
          src="https://www.mymilestone.com/blog/wp-content/uploads/2017/12/forecast-1.jpg"
        />
        <h3>
          Austin, Texas
        </h3>
        <span
          className="location-detail"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the
        </span>
      </div>
      <div
        className="location"
        key="San Jose, California"
      >
        <img
          alt="San Jose, California"
          className="location-image"
          src="https://www.valleyinnsanjose.com/gallery-images/properties/5/0/1/1540545718_attractions_1.jpg"
        />
        <h3>
          San Jose, California
        </h3>
        <span
          className="location-detail"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the
        </span>
      </div>
      <div
        className="location"
        key="Tokyo, Japan"
      >
        <img
          alt="Tokyo, Japan"
          className="location-image"
          src="https://www.telegraph.co.uk/content/dam/Travel/Destinations/Asia/Japan/Tokyo/tokyo-city-scape.jpg?imwidth=450"
        />
        <h3>
          Tokyo, Japan
        </h3>
        <span
          className="location-detail"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the
        </span>
      </div>
    </div>
  </div>
</div>
`;
